File: VB/vbnet/Get_Data/URL_Generator.vb
' *
' * Bluepay VB.NET Sample code.
' *
' * This code sample runs a report that grabs a single transaction
' * from the BluePay gateway based on certain criteria.
' * See comments below on the details of the report.
' * If using TEST mode, only TEST transactions will be returned.
' *

Imports System
Imports vbnet.BPVB

Namespace Sample

    Public Class URLGeneratorMethodSample

        Public Shared Sub run()

            Dim accountID As String = "Merchant's Account ID Here"
            Dim secretKey As String = "Merchant's Secret Key Here"
            Dim mode As String = "TEST" 

            Dim testURL As BluePay = New BluePay(
                accountID,
                secretKey,
                mode
            )
            
            Dim generatedURL As String = testURL.generateURL(
              merchantName:= "Test Merchant",
              returnURL:= "www.google.com",
              transactionType:= "SALE",
              acceptDiscover:= "Yes",
              acceptAmex:= "Yes",
              amount:= "99.99", 
              protectAmount:= "Yes",
              rebilling:= "Yes",
              rebProtect:= "Yes", 
              rebAmount:= "50", 
              rebCycles:= "12", 
              rebStartDate:= "1 MONTH", 
              rebFrequency:= "1 MONTH", 
              customID1:= "MyCustomID1.1234", 
              protectCustomID1:= "Yes", 
              customID2:= "MyCustomID2.12345678910", 
              protectCustomID2:= "Yes", 
              paymentTemplate:= "mobileform01", 
              receiptTemplate:= "defaultres2", 
              receiptTempRemoteURL:= ""
            )

            Console.WriteLine("Hosted Payment Form URL: " + generatedURL)
    
        End Sub
    End Class
End Namespace

File: Ruby/samples/Get_Data/URL_Generator.rb
require_relative "bluepay.rb"


ACCOUNT_ID = "Merchant's Account ID Here"
SECRET_KEY = "Merchant's Secret Key Here"
MODE = "TEST"

test_url = BluePay.new(
  account_id: ACCOUNT_ID,  
  secret_key: SECRET_KEY,  
  mode: MODE
)

generated_url = test_url.generate_url(
merchant_name: "Test Merchant",
return_url: "www.google.com",
transaction_type: "SALE", 
accept_discover: "Yes", 
accept_amex: "Yes", 
amount: "99.99", 
protect_amount: "Yes",
rebilling: "Yes",
reb_protect: "Yes", 
reb_amount: "50", 
reb_cycles: "12",
reb_start_date: "1 MONTH",
reb_frequency: "1 MONTH",
custom_id: "MyCustomID1.1234",
protect_custom_id: "Yes",
custom_id2: "MyCustomID2.12345678910",
protect_custom_id2: "Yes",
payment_template: "mobileform01", 
receipt_template: "defaultres2", 
receipt_temp_remote_url: ""
 )

puts "Hosted Payment Form URL: " + generated_url